internal/command/views/view.go

```diff
@@ -136,6 +151,23 @@ func (v *View) Diagnostics(diags tfdiags.Diagnostics) {
 	}
 }
 
+// HasErrors accepts a set of Diagnostics and interrogates both the View and Diagnostics returning whether an error
+// has occurred or not.
+func (v *View) HasErrors(diags tfdiags.Diagnostics) bool {
+	if v.PedanticWarningFlagged {
+		// If a pedantic warning has been flagged then we report the view is in error state.
+		// PedanticWarningFlagged can be set from legacy UI / diagnostics rendering and new view diagnostics rendering.
```

Comment:
I don't think we need to track PedanticWarningFlagged at all given the latest iteration of this PR?

I think we can remove it and simplify quite a lot of code?  Curious on your thoughts @bunniseng.

Reply:
Digging a bit further into this, it appears you are trying to handle:
```
internal/backend/backend.go:            o.View.Diagnostics(diags)
internal/backend/local/backend.go:                      view.Diagnostics(diags)
internal/backend/local/backend_plan.go:                 op.View.Diagnostics(diags)
internal/backend/local/backend_refresh.go:                      op.View.Diagnostics(diags)
internal/backend/local/backend_apply.go:                        op.View.Diagnostics(diags)
internal/backend/local/backend_apply.go:                                op.View.Diagnostics(diags)
internal/backend/local/backend_apply.go:        op.View.Diagnostics(diags)
```

Is that correct?

Reply:
After reviewing this PR, I see only two places where legacy `Warn` function call is actually should be checked for pedantic (feel free to point me if I miss something):

* [warning in cloud backend](https://github.com/bunniseng/opentofu/blob/feature/757_warnings_to_errors_new/internal/cloud/backend_run_warning.go#L40)
* [warning in init command](https://github.com/bunniseng/opentofu/blob/feature/757_warnings_to_errors_new/internal/command/init.go#L880)

Can we handle these cases separately somehow (put messages as diags?) and then remove `PedanticWarningFlagged`? What do you all think?

Reply:
Just a small note on the [Warn call in cloud command](https://github.com/bunniseng/opentofu/blob/feature/757_warnings_to_errors_new/internal/command/cloud.go#L95). We have `return 1` after that call, so exit code is always non-zero in this case.

Reply:
I am in favor of removing it and refactoring those few Warn calls. I might be missing some of the command package quirks so let's see what @cam72cam thinks about the whole thread.